app/src/main/java/com/example/inventory/ui/item/ItemEntryScreen.kt

```diff
@@ -0,0 +1,177 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Button
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.example.inventory.InventoryTopAppBar
+import com.example.inventory.R
+import com.example.inventory.ui.AppViewModelProvider
+import com.example.inventory.ui.navigation.NavigationDestination
+import com.example.inventory.ui.theme.InventoryTheme
+import kotlinx.coroutines.launch
+import java.util.Currency
+import java.util.Locale
+
+object ItemEntryDestination : NavigationDestination {
+    override val route = "item_entry"
+    override val titleRes = R.string.item_entry_title
+}
+
+@Composable
+fun ItemEntryScreen(
+    navigateBack: () -> Unit,
+    onNavigateUp: () -> Unit,
+    modifier: Modifier = Modifier,
+    canNavigateBack: Boolean = true,
+    viewModel: ItemEntryViewModel = viewModel(factory = AppViewModelProvider.Factory)
+) {
+    val coroutineScope = rememberCoroutineScope()
+    Scaffold(topBar = {
+        InventoryTopAppBar(
+            title = stringResource(ItemEntryDestination.titleRes),
+            canNavigateBack = canNavigateBack,
+            navigateUp = onNavigateUp
+        )
+    }) { innerPadding ->
+        ItemEntryBody(
+            itemUiState = viewModel.itemUiState,
+            onItemValueChange = viewModel::updateUiState,
+            onSaveClick = {
+                // Note: If the user rotates the screen very fast, the operation may get cancelled
+                // and the item may not be saved in the Database. This is because when config
+                // change occurs, the Activity will be recreated and the rememberCoroutineScope will
+                // be cancelled - since the scope is bound to composition.
+                coroutineScope.launch {
+                    viewModel.saveItem()
+                    navigateBack()
+                }
+            },
+            modifier = modifier.padding(innerPadding)
+        )
+    }
+}
+
+@Composable
+fun ItemEntryBody(
+    itemUiState: ItemUiState,
+    onItemValueChange: (ItemDetails) -> Unit,
+    onSaveClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
```

Comment:
pass these in from the parent

Reply:
This is tricky, could you try doing from your end?

Reply:
ping me about this tomorrow

Reply:
did we end up resolving this?